app/Http/Controllers/SquareController.php

```diff
@@ -0,0 +1,206 @@
+        $orderRequest->setIdempotencyKey($payment->unique_id);
+
+        $checkoutRequest = new CreateCheckoutRequest($payment->unique_id, $orderRequest);
+        $checkoutRequest->setMerchantSupportEmail('treasurer@robojackets.org');
+        $checkoutRequest->setPrePopulateBuyerEmail($user->gt_email);
+        $checkoutRequest->setRedirectUrl(route('pay.complete'));
+
+        $square = new SquareClient([
+            'accessToken' => config('square.access_token'),
+            'environment' => config('square.environment'),
+        ]);
+
+        $checkoutApi = $square->getCheckoutApi();
+        $checkoutResponse = $checkoutApi->createCheckout(config('square.location_id'), $checkoutRequest);
+
+        if (! $checkoutResponse->isSuccess()) {
+            Log::error(self::class.' Error creating checkout - '.json_encode($checkoutResponse->getErrors()));
+
+            return view(
+                'square.error',
+                [
+                    'message' => 'We could not contact Square to begin your payment.',
+                ]
+            );
+        }
+
+        $checkout = $checkoutResponse->getResult()->getCheckout();
+
+        $payment->checkout_id = $checkout->getId();
+        $payment->order_id = $checkout->getOrder()->getId();
+        $payment->save();
+
+        return redirect($checkout->getCheckoutPageUrl());
+    }
+
+    public function complete(SquareCompleteRequest $request)
+    {
+        $payment = Payment::where('id', $request->input('referenceId'))->firstOrFail();
+
+        if ($payment->checkout_id !== $request->input('checkoutId')) {
+            return view(
+                'square.error',
+                [
+                    'message' => 'We could not match your payment in our database.',
+                ]
+            );
+        }
+
+        $square = new SquareClient([
+            'accessToken' => config('square.access_token'),
+            'environment' => config('square.environment'),
+        ]);
+
+        $ordersApi = $square->getOrdersApi();
+
+        $retrieveOrderResponse = $ordersApi->retrieveOrder($payment->order_id);
+
+        if (! $retrieveOrderResponse->isSuccess()) {
+            Log::error(self::class.' Error retrieving order - '.json_encode($retrieveOrderResponse->getErrors()));
+
+            return view(
+                'square.error',
+                [
+                    'message' => 'We could not contact Square to finalize your payment.',
+                ]
+            );
+        }
+
+        $order = $retrieveOrderResponse->getResult()->getOrder();
+
+        switch ($order->getState()) {
+            case OrderState::COMPLETED:
+                $tender = $order->getTenders()[0];
+                $payment->amount = $tender->getAmountMoney()->getAmount() / 100;
+                if (null !== $tender->getProcessingFeeMoney()) {
+                    $payment->processing_fee = $tender->getProcessingFeeMoney()->getAmount() / 100;
+                }
```

Comment:
It seems like the processing fee is not available at this point, but I figured it was worth checking.